### ISSUE TITLE
Raiden/SanitySwap - request is conflicting or otherwise invalid

### ISSUE THREAD

**Issue by offerm** [ExchangeUnion/xud#957]
XUD tries to swap with a partner:
```
{
  "node": "2b88992d",
  "registry_address": "0x81E09B9Ba95c9D1fcC6Be59D7798eFB8E1EF342E",
  "token_address": "0x4c354C76d5f73A63a90Be776897DC81Fb6238772",
  "target_address": "0x7ed0299Fa1ADA71D10536B866231D447cDFa48b9",
  "amount": 1,
  "payment_identifier": 1697660191604947,
  "secret": null,
  "secret_hash": "0x8118f1085c9f7195d32a71ad1ebc28c49cb758f36514995acbf5945d2670e26f",
  "event": "Initiating payment",
  "logger": "raiden.api.rest",
  "level": "debug",
  "timestamp": "2019-05-08 20:55:54.274534"
}
```

Raiden now response fast back to XUD with an error:
```
{
  "node": "2b88992d",
  "raiden_events": [
    "{\"payment_network_identifier\": \"0x81E09B9Ba95c9D1fcC6Be59D7798eFB8E1EF342E\", \"token_network_identifier\": \"0xeEC30Bf645f1B7E5607450F6291bAfBdD22B1892\", \"identifier\": \"16
97660191604947\", \"target\": \"0x7ed0299Fa1ADA71D10536B866231D447cDFa48b9\", \"reason\": \"Target requested the secret but secret is unknown to the Initiator\", \"_type\": \"raiden.tr
ansfer.events.EventPaymentSentFailed\", \"_version\": 0}"
  ],
  "event": "Raiden events",
  "logger": "raiden.raiden_service",
  "level": "debug",
  "greenlet_name": "GMatrixClient._sync user_id:@0x2b88992ded5c96aa7eaa9cfe1ae52350df7dc5df:raidentransport.exchangeunion.com sync_token:s47340_225203_0_1_66_1_1_1076_1",
  "timestamp": "2019-05-08 20:55:55.239066"
}
```

XUD log does not show the error but this:
```
08/05/2019 20:55:55.245 [RAIDEN] error: got exception from RaidenClient.tokenPayment:: 5.4 - request is conflicting or otherwise invalid
08/05/2019 20:55:55.246 [SWAPS] warn: got payment error during sanity swap with 02b66438730d1fcdf4a4ae5d3d73e847a272f160fee2938e132b52cab0a0d9cfc6 for WETH using rHash 8118f1085c9f7195
d32a71ad1ebc28c49cb758f36514995acbf5945d2670e26f: request is conflicting or otherwise invalid
```

- There should be no Exception
- Should show the error to make it clear what the problem is
- Should not try Sanity swap if there is no outgoing capacity.

**Comment by kilrau** [comment]
addon:
* Should print error in xud.log (e.g. errors from raiden (HTTP 409) are not being printed into xud.log atm)

**Comment by sangaman** [comment]
>     There should be no Exception
>     Should show the error to make it clear what the problem is
>     Should not try Sanity swap if there is no outgoing capacity.

For the first point, what do you mean exactly? Removing the `08/05/2019 20:55:55.245 [RAIDEN] error: got exception from RaidenClient.tokenPayment:: 5.4 - request is conflicting or otherwise invalid` line from the logs? The exception is just part of the flow for when a raiden payment fails.

For the last point, that's probably a good idea but I think should be handled separately from the logging enhancements, possibly for another issue too.

I'll open a PR for the second point and possibly modify it after clarification on the first point.

**Comment by offerm** [comment]
For the first point - There was no `exception` when calling raiden, right? we just got HTTP 409 back. The exception is an internal mechanism that we are using to inform the caller. Don't think it should get to the log like this. Not a big issue.

The main comment here is about the second point - raiden is giving us back a detailed error and we should show that and not a text that we created.  In the above case
```
\"reason\": \"Target requested the secret but secret is unknown to the Initiator\"
```
